label_studio/io_storages/gcs/utils.py

```diff
@@ -0,0 +1,231 @@
+                GCS._credentials_cache[cache_key] = credentials
+            out = {
+                "service_account_email": credentials.service_account_email,
+                "access_token": credentials.token,
+                "credentials": credentials
+            }
+        except DefaultCredentialsError as exc:
+            logger.error(f"Label studio couldn't load default GCS credentials from env. {exc}",
+                         exc_info=True)
+            out = {}
+        return out
+
+    @classmethod
+    def generate_http_url(
+        cls,
+        url: str,
+        google_application_credentials: Union[str, dict] = None,
+        google_project_id: str = None,
+        presign_ttl: int = 1
+    ) -> str:
+        """
+        Gets gs:// like URI string and returns presigned https:// URL
+        :param url: input URI
+        :param google_application_credentials:
+        :param google_project_id:
+        :param presign_ttl: Presign TTL in minutes
+        :return: Presigned URL string
+        """
+        r = urlparse(url, allow_fragments=False)
+        bucket_name = r.netloc
+        blob_name = r.path.lstrip('/')
+        """Generates a v4 signed URL for downloading a blob.
+
+        Note that this method requires a service account key file. You can not use
+        this if you are using Application Default Credentials from Google Compute
+        Engine or from the Google Cloud SDK.
+        """
+        # bucket_name = 'your-bucket-name'
+        # blob_name = 'your-object-name'
+
+        client = cls.get_client(
+            google_application_credentials=google_application_credentials,
+            google_project_id=google_project_id
+        )
+        bucket = client.get_bucket(bucket_name)
+        blob = bucket.blob(blob_name)
+
+        url = blob.generate_signed_url(
+            version="v4",
+            # This URL is valid for 15 minutes
+            expiration=timedelta(minutes=presign_ttl),
+            # Allow GET requests using this URL.
+            method="GET",
+            **cls._get_signing_kwargs(google_application_credentials)
+        )
+
+        logger.debug('Generated GCS signed url: ' + url)
+        return url
+
+    @classmethod
+    def iter_images_base64(cls, client, bucket_name, max_files):
+        for image in cls.iter_blobs(client, bucket_name, max_files):
+            yield GCS.read_base64(image)
+
+    @classmethod
+    def iter_images_filename(cls, client, bucket_name, max_files):
+        for image in cls.iter_blobs(client, bucket_name, max_files):
+            yield image.name
+
+    @classmethod
+    def get_uri(cls, bucket_name, key):
+        return f'gs://{bucket_name}/{key}'
+
+    @classmethod
+    def _try_read_json(cls, blob_str):
+        try:
+            data = json.loads(blob_str)
+        except ValueError:
+            logger.error(f"Can't parse JSON from {blob_str}")
+            return
+        return data
+
+    @classmethod
+    def read_file(
```

Comment:
Will this be used for reading files from a bucket for creating a new Dataset ? Or is this just for normal LSE projects?

Reply:
Not sure if we use universal storage adapter from dataflow